openmoltools/amber.py

```diff
@@ -223,7 +253,13 @@ def run_antechamber(molecule_name, input_filename, charge_method="bcc", net_char
     if frcmod_filename is None:
         frcmod_filename = molecule_name + '.frcmod'
 
-    cmd = "antechamber -i %s -fi mol2 -o %s -fo mol2 -s 2" % (input_filename, gaff_mol2_filename)
+    #Use temporary directory to do this to avoid issues with spaces in filenames, etc.
+    tempdir = tempfile.mkdtemp()
+    startdir = os.getcwd()
+    shutil.copy( input_filename, os.path.join( tempdir, 'in.mol2') )
+    os.chdir( tempdir )
```

Comment:
Its how the enter_temp_directory in mdtraj.utils works
On Jun 25, 2015 7:51 PM, "John Chodera" <notifications@github.com> wrote:

> In openmoltools/amber.py
> <https://github.com/choderalab/openmoltools/pull/138#discussion_r33325223>
> :
>
> > @@ -223,7 +253,13 @@ def run_antechamber(molecule_name, input_filename, charge_method="bcc", net_char
> >      if frcmod_filename is None:
> >          frcmod_filename = molecule_name + '.frcmod'
> >
> > -    cmd = "antechamber -i %s -fi mol2 -o %s -fo mol2 -s 2" % (input_filename, gaff_mol2_filename)
> > +    #Use temporary directory to do this to avoid issues with spaces in filenames, etc.
> > +    tempdir = tempfile.mkdtemp()
> > +    startdir = os.getcwd()
> > +    shutil.copy( input_filename, os.path.join( tempdir, 'in.mol2') )
> > +    os.chdir( tempdir )
>
> Here's a great little summary:
> http://effbot.org/zone/python-with-statement.htm
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/choderalab/openmoltools/pull/138/files#r33325223>.
>